lib/awful/keyboard.lua

```diff
@@ -0,0 +1,77 @@
+---------------------------------------------------------------------------
+--- Utilities related to the keyboard and text entry.
+--
+-- @author Emmanuel Lepage Vallee &lt;elv1313@gmail.com&gt;
+-- @copyright 2018 Emmanuel Lepage Vallee
+-- @module awful.keyboard
+---------------------------------------------------------------------------
+local capi = {root = root, client = client}
+
+local module = {}
+
+-- The hash order isn't consistent when the content changes, but it is
+-- constant when the table doesn't change.
```

Comment:
I have no idea what this comment talks about. I guess it refers to the iteration order with `pairs`, but I do not see where this is important (and I'd be surprised if the Lua documentation actually gives this guarantee, but whatever...).